lib/internal/child_process/serialization.js

```diff
@@ -51,29 +55,47 @@ class ChildProcessDeserializer extends v8.DefaultDeserializer {
 //   (aka 'advanced')
 const advanced = {
   initMessageChannel(channel) {
-    channel[kMessageBuffer] = Buffer.alloc(0);
+    channel[kMessageBuffer] = [];
+    channel[kMessageBufferSize] = 0;
     channel.buffering = false;
   },
 
   *parseChannelMessages(channel, readData) {
     if (readData.length === 0) return;
 
-    let messageBuffer = Buffer.concat([channel[kMessageBuffer], readData]);
-    while (messageBuffer.length > 4) {
-      const size = messageBuffer.readUInt32BE();
-      if (messageBuffer.length < 4 + size) {
-        break;
-      }
+    ArrayPrototypePush(channel[kMessageBuffer], readData);
+    channel[kMessageBufferSize] += readData.length;
+
+    // Index 0 should always be present because we just pushed data into it.
+    let messageBufferHead = channel[kMessageBuffer][0];
+    while (messageBufferHead.length >= 4) {
+      // We call `readUInt32BE` manually here, because this is faster than first converting
+      // it to a buffer and using `readUInt32BE` on that.
+      const fullMessageSize = ReflectApply(readUInt32BE, messageBufferHead, [0]) + 4;
```

Comment:
Drive-by suggestion: read the int inline, that's both faster and shorter (and arguably easier to read)
```suggestion
      const b = messageBufferHead;
      const fullMessageSize = b[0] * 0x1000000 + b[1] * 65536 + b[2] * 256 + b[3];
```